From: Hans van Kranenburg <hans@knorrie.org>
Date: Sat, 9 Feb 2019 17:27:26 +0100
Subject: sysconfig.xencommons.in: Strip and debianize

Strip all options that are for stuff we don't ship, which is 1)
xenstored as stubdom and 2) the new options for oom score and open file
descriptor limit, which would not have any effect, because we're
shipping different init scripts... :|

It seems useful to give the user the option to revert to xenstored
instead of the default oxenstored if they really want.

Signed-off-by: Hans van Kranenburg <hans@knorrie.org>
Acked-by: Ian Jackson <ijackson@chiark.greenend.org.uk>
---
 tools/hotplug/Linux/init.d/sysconfig.xencommons.in | 82 +---------------------
 1 file changed, 3 insertions(+), 79 deletions(-)

diff --git a/tools/hotplug/Linux/init.d/sysconfig.xencommons.in b/tools/hotplug/Linux/init.d/sysconfig.xencommons.in
index 433e484..e948543 100644
--- a/tools/hotplug/Linux/init.d/sysconfig.xencommons.in
+++ b/tools/hotplug/Linux/init.d/sysconfig.xencommons.in
@@ -6,104 +6,28 @@
 #XENCONSOLED_TRACE=[none|guest|hv|all]
 
 ## Type: string
-## Default: daemon
+## Default: oxenstored
 #
-# Select type of xentore service.
-#
-# This can be either of:
-#  * daemon
-#  * domain
-#
-# Changing this requires a reboot to take effect.
-#
-#XENSTORETYPE=daemon
-
-## Type: string
-## Default: xenstored
-#
-# Select xenstore implementation, this can be either
-# of these below.
-# Only evaluated if XENSTORETYPE is "daemon".
-#
-# This can be either of:
+# Select xenstore implementation. This can be either of:
 #  * @sbindir@/oxenstored
 #  * @sbindir@/xenstored
 #
 # Changing this requires a reboot to take effect.
 #XENSTORED=@XENSTORED@
 
-## Type: string
-## Default: unlimited
-#
-# Select maximum number of file descriptors xenstored is allowed to have
-# opened at one time.
-# For each HVM domain xenstored might need up to 5 open file descriptors,
-# PVH and PV domains will require up to 3 open file descriptors. Additionally
-# 20-30 file descriptors will be opened for internal uses.
-# The specified value (including "unlimited") will be capped by the contents
-# of /proc/sys/fs/nr_open if existing.
-# Only evaluated if XENSTORETYPE is "daemon".
-#XENSTORED_MAX_OPEN_FDS=unlimited
-
 ## Type: string
 ## Default: ""
 #
 # Additional commandline arguments to start xenstored,
 # like "--trace-file @XEN_LOG_DIR@/xenstored-trace.log"
-# See "@sbindir@/xenstored --help" for possible options.
-# Only evaluated if XENSTORETYPE is "daemon".
+# See "@sbindir@/oxenstored --help" for possible options.
 XENSTORED_ARGS=
 
 ## Type: string
 ## Default: Not defined, tracing off
 #
 # Log xenstored messages
-# Only evaluated if XENSTORETYPE is "daemon".
 #XENSTORED_TRACE=[yes|on|1]
 
-## Type: integer
-## Default: 50
-#
-# Percentage of dom0 memory size the xenstore daemon can use before the
-# OOM killer is allowed to kill it.
-# The specified value is multiplied by -10 and echoed to
-# /proc/PID/oom_score_adj.
-#XENSTORED_OOM_MEM_THRESHOLD=50
-
-## Type: string
-## Default: @LIBEXEC@/boot/xenstore-stubdom.gz
-#
-# xenstore domain kernel.
-# Only evaluated if XENSTORETYPE is "domain".
-#XENSTORE_DOMAIN_KERNEL=@LIBEXEC@/boot/xenstore-stubdom.gz
-
-## Type: integer
-## Default: 8
-#
-# xenstore domain memory size in MiB.
-# Only evaluated if XENSTORETYPE is "domain".
-#XENSTORE_DOMAIN_SIZE=8
-
-## Type: string
-## Default: not set, no autoballooning of xenstore domain
-#
-# Maximum xenstore domain memory size. Can be specified as:
-# - plain integer value for max size in MiB
-# - fraction of host memory, e.g. 1/100
-# - combination of both in form of <val>:<frac> (e.g. 8:1/100), resulting
-#   value will be the higher of both specifications
-# Only evaluated if XENSTORETYPE is "domain".
-#XENSTORE_MAX_DOMAIN_SIZE=
-
-## Type: string
-## Default: ""
-#
-# Additional arguments for starting the xenstore domain.
-# Only evaluated if XENSTORETYPE is "domain".
-XENSTORE_DOMAIN_ARGS=
-
-# qemu path
-#QEMU_XEN=@qemu_xen_path@
-
 # Dom0 UUID
 #XEN_DOM0_UUID=00000000-0000-0000-0000-000000000000
